crates/js-component-bindgen/src/lib.rs

```diff
@@ -172,3 +172,35 @@ pub fn dealias(resolve: &Resolve, mut id: TypeId) -> TypeId {
         }
     }
 }
+
+/// Check if an item (usually some form of [`WorldItem`]) should be allowed through the feature gate
+/// of a given package.
+fn feature_gate_allowed(
+    resolve: &Resolve,
+    package: &Package,
+    stability: &Stability,
+) -> Result<bool> {
+    let Some(package_version) = package.name.version.as_ref() else {
+        // If the package version is missing (we're likely dealing with an unresolved package)
+        // and we can't really check much.
+        return Ok(true);
+    };
+
+    Ok(match stability {
+        Stability::Unknown => true,
+        Stability::Stable { since, .. } => {
```

Comment:
We need to also check the `feature` here as well.

Reply:
If I'm understanding this right you mean to use that to determine visibility right? Originally I thought so too -- but after the [discussion upstream](https://github.com/WebAssembly/component-model/issues/382) the consensus was that once a feature is *stable*, the written "feature" is actually just descriptive. 

Ideally I'd just reuse the function from [the `wasm-tools` PR](https://github.com/bytecodealliance/wasm-tools/pull/1689) in `witparser` here but it's not released just yet.

Reply:
Also, with regards to #1, trying to refer to a future package (i.e. `@since( version = 0.2.1)`, when the package is `0.2.0` is actually a bug -- wasmparser wouldn't output that, so I could add a test for the failure but not that the specific items were missing.

Reply:
Ah yep, so basically I think the *intended* progression is that once a feature goes from `@unstable(feature = ...)` to `@since(version = X.X.X)` it is "stable" and is basically always included -- the "feature" annotation doesn't really do anything at that point.